cypress/component/ResultContainer.cy.tsx

```diff
@@ -0,0 +1,66 @@
+import ResultContainer from '../../src/components/ResultContainer';
+import { protectedResponse2 } from '../fixtures/mocked-responses';
+
+describe('ResultContainer', () => {
+  it('Displays a set of Result Cards, select all checkbox, disabled download result buttons, summary stats, and how to get data modal button', () => {
+    cy.mount(<ResultContainer result={protectedResponse2} />);
+    cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset"]').should('be.visible');
+    cy.get('[data-cy="card-http://neurobagel.org/vocab/not-so-cool-dataset"]').should('be.visible');
+    cy.get('[data-cy="select-all-checkbox"]').should('be.visible');
+    cy.get('[data-cy="summary-stats"]')
+      .should('be.visible')
+      .should('contain', 'Summary stats: 2 datasets, 4 subjects');
+    cy.get('[data-cy="participant-level-download-results-button"]')
+      .should('be.visible')
+      .should('be.disabled');
+    cy.get('[data-cy="dataset-level-download-results-button"]')
+      .should('be.visible')
+      .should('be.disabled');
+    cy.get('[data-cy="how-to-get-data-modal-button"]').should('be.visible');
+  });
+  it('Selecting a dataset should enable the download result buttons', () => {
+    cy.mount(<ResultContainer result={protectedResponse2} />);
+    cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-checkbox"] input').check();
+    cy.get('[data-cy="participant-level-download-results-button"]')
+      .should('be.visible')
+      .should('not.be.disabled');
+    cy.get('[data-cy="dataset-level-download-results-button"]')
+      .should('be.visible')
+      .should('not.be.disabled');
+  });
+  it('Selecting/unselecting select all datasets checkbox should check/uncheck all dataset cards', () => {
+    cy.mount(<ResultContainer result={protectedResponse2} />);
+    cy.get('[data-cy="select-all-checkbox"] input').check();
```

Comment:
similar to above, this mixes the view and the logic test. Another way to describe the test is: 

if my selectedDownloads list is 
- empty, all dataset cards should be unchecked
- has some datasets, then these dataset cards should be checked

how selectedDownloads is set via the UI is not related

Reply:
I see but I think the only way to manipulate the `selectedDwonloads`  is through the UI. If we want to mock the `download`, we have to change the component's logic and take the default value as a prop.

Reply:
OK, let's not deal with this here / in this PR. Can think a bit more about this later